pkg/dispatcher/dispatcher.go

```diff
@@ -180,25 +190,33 @@ func (d *Dispatcher) dispatch(ctx context.Context, msg amqp.Delivery, ceClient c
 	statusCode, isSuccess := getStatus(ctx, result)
 	if statusCode != -1 {
 		args := &dispatcher.ReportArgs{EventType: event.Type()}
-		if err := d.Reporter.ReportEventCount(args, statusCode); err != nil {
+		if err = d.Reporter.ReportEventCount(args, statusCode); err != nil {
 			logging.FromContext(ctx).Errorf("Something happened: %v", err)
+
+			msg.Headers["knativeerrordest"] = d.SubscriberURL
```

Comment:
And you can remove it from here, this is just knative related so if it fails it shouldn't affect the user